scio-jdbc/src/main/scala/com/spotify/scio/jdbc/JdbcIO.scala

```diff
@@ -66,7 +64,7 @@ object JdbcIO {
 final case class JdbcSelect[T: Coder](readOptions: JdbcReadOptions[T]) extends JdbcIO[T] {
```

Comment:
IMHO we've not defined the constructor for `JdbcIOs` properly. Here we should only have data that allows to identify the target destination (required to distinguish the mocked IO basically), so the connection option and the query.

All the other param should be passed as `ReadP` or `WriteP`